lib/init.coffee

```diff
@@ -1,8 +1,50 @@
+{BufferedProcess, CompositeDisposable} = require 'atom'
+
 module.exports =
+
   config:
     javaExecutablePath:
       type: 'string'
-      default: ''
+      title: 'Path to the javac executable'
+      default: '/usr/bin/javac'
 
   activate: ->
-    console.log 'activate linter-javac'
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe 'linter-javac.javaExecutablePath',
+      (newValue) =>
+        @javaExecutablePath = newValue
+
+  deactivate: ->
+    @subscriptions.dispose()
+
+  provideLinter: ->
+    grammarScopes: ['source.java']
+    scope: 'file'
+    lintOnFly: true
+    lint: (textEditor) =>
+      return new Promise (resolve, reject) =>
+        filePath = textEditor.getPath()
+        messages = []
+        process = new BufferedProcess
```

Comment:
Simply because that is what was suggested by the example in the linter wiki: https://github.com/AtomLinter/Linter/wiki/Migrating-to-the-new-API (I'm new to Atom packages).  :)  Is there a better option?